src/v/cluster/health_monitor_backend.cc

```diff
@@ -674,4 +688,46 @@ std::chrono::milliseconds health_monitor_backend::max_metadata_age() {
     return config::shard_local_cfg().health_monitor_max_metadata_age();
 }
 
+ss::future<cluster_health_overview>
+health_monitor_backend::get_cluster_health_overview(
+  model::timeout_clock::time_point deadline) {
+    auto ec = co_await maybe_refresh_cluster_health(
+      force_refresh::no, deadline);
```

Comment:
should cluster health reports always be served by the controller? that would also be an implicit health check: client can at least connect to the controller, but it would also mean that clients get consistent answers. if clients can contact any node then a sequence of requests might produce a weird series of states?

Reply:
exactly, i assumed that if we can not refresh cluster is in bad condition.

Reply:
hmm I'm not very familiar with this code so please tell me if I have some wrong assumptions.

but I think if the cluster was healthy, but we couldn't refresh, then we may have `cluster_health_overview` with no nodes down, all nodes have leaders, but `is_healthy` is false, which may be confusing.

Maybe we should add a field like `obsolete cluster state` to the `cluster_health_overview`?